### PR TITLE
Handle integration exceptions when automation YAML is malformed

### LINKED ISSUE #19694: When device is deleted, automation UI fails to handle references to that device cleanly
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Deleting a device causes the Automation Editor's Device Action values to become empty. When this happens (usually without you noticing) and you try to save your automation in the editor, you receive an error message "Message malformed: Integration '' not found".

image

# UI
image

# YAML
image

# Browser Console
image

# API Response
image

### Describe the behavior you expected

I vaguely recall that the toast used to say the actual error, such as "Unknown device 'e9eee6046b80aabf468dc51c90a26297'" or "Device not found" but that does not happen anymore (despite that both of those show in the browser console).

If we can avoid sending malformed data to HA Core that would always be better.

### Steps to reproduce the issue

1. Create automation with device action and save
2. Delete that device itself (not the automation entry)
3. View the automation device action

### What version of Home Assistant Core has the issue?

core-2024.2.0b7

### What was the last working version of Home Assistant Core?

Prior to 2020

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

```txt
Before the device was deleted, device_id, domain and entity_id were all populated.
```

### Problem-relevant frontend configuration

```yaml
alias: Garage - EV Schedule
description: ""
trigger:
  - platform: time
    at: "21:01:00"
    id: time_start
    alias: Begin Off Peak (TOD)
  - platform: time
    at: "06:59:00"
    id: time_stop
    alias: End Off-Peak (TOD)
  - alias: Begin solar (producing more than 1.8 kW)
    platform: numeric_state
    entity_id:
      - sensor.garage_solar_inverter_pv_power
    for:
      hours: 0
      minutes: 5
      seconds: 0
    id: solar_start
    above: 1.8
  - alias: End solar (producing less 1.8 kW)
    platform: numeric_state
    entity_id:
      - sensor.garage_solar_inverter_pv_power
    for:
      hours: 0
      minutes: 5
      seconds: 0
    below: 1.8
    id: solar_stop_fixed
  - alias: End solar (producing less than EV is consuming)
    platform: numeric_state
    entity_id:
      - sensor.garage_solar_inverter_pv_power
    for:
      hours: 0
      minutes: 5
      seconds: 0
    below: sensor.garage_ev_current_power
    id: solar_stop_dynamic
condition:
  - condition: numeric_state
    entity_id: zone.home
    above: 0
action:
  - choose:
      - conditions:
          - condition: trigger
            id:
              - solar_start
              - time_start
        sequence:
          - device_id: ""
            domain: ""
            entity_id: ""
          - device_id: ""
            domain: ""
            entity_id: ""
      - conditions:
          - condition: trigger
            id:
              - solar_stop_fixed
              - solar_stop_dynamic
        sequence:
          - device_id: ""
            domain: ""
            entity_id: ""
          - device_id: ""
            domain: ""
            entity_id: ""
      - conditions:
          - condition: trigger
            id:
              - time_stop
        sequence:
          - device_id: ""
            domain: ""
            entity_id: ""
          - device_id: ""
            domain: ""
            entity_id: ""
mode: parallel
max: 10
```

### Javascript errors shown in your browser console/inspector

```txt
ha-automation-editor.ts:494  Uncaught (in promise) Objectcode: "home_assistant_error"message: "Unknown device 'e9eee6046b80aabf468dc51c90a26297'"[[Prototype]]: Object
ha-device-automation-picker.ts:160  Uncaught (in promise) Objectcode: "not_found"message: "Device not found"[[Prototype]]: Object
ha-device-automation-picker.ts:160  Uncaught (in promise) Objectcode: "not_found"message: "Device not found"[[Prototype]]: Object
ha-device-automation-picker.ts:160  Uncaught (in promise) Objectcode: "not_found"message: "Device not found"[[Prototype]]: Object
ha-device-automation-picker.ts:160  Uncaught (in promise) Objectcode: "not_found"message: "Device not found"[[Prototype]]: Object
ha-device-automation-picker.ts:160  Uncaught (in promise) Objectcode: "not_found"message: "Device not found"[[Prototype]]: Object
ha-device-automation-picker.ts:160  Uncaught (in promise) Objectcode: "not_found"message: "Device not found"[[Prototype]]: Object
90[Violation] Added non-passive event listener to a scroll-blocking <some> event. Consider marking event handler as 'passive' to make the page more responsive. See <URL>
[Violation] Forced reflow while executing JavaScript took 35ms
fetch-with-auth.ts:20 
        
        
        POST https://xxx/api/config/automation/config/1697489596989 400 (Bad Request)
U @ fetch-with-auth.ts:20
await in U (async)
(anonymous) @ hass-call-api.ts:71
callApi @ connection-mixin.ts:155
h @ automation.ts:353
value @ ha-automation-editor.ts:723
handleEvent @ lit-html.ts:2018
ha-automation-editor.ts:737  Uncaught (in promise) {error: 'Response error: 400', status_code: 400, body: {…}}
value @ ha-automation-editor.ts:737
await in value (async)
handleEvent @ lit-html.ts:2018
fetch-with-auth.ts:20 
        
        
        POST https://xxx/api/config/automation/config/1697489596989 400 (Bad Request)
U @ fetch-with-auth.ts:20
(anonymous) @ hass-call-api.ts:71
callApi @ connection-mixin.ts:155
h @ automation.ts:353
value @ ha-automation-editor.ts:723
handleEvent @ lit-html.ts:2018
ha-automation-editor.ts:737  Uncaught (in promise) {error: 'Response error: 400', status_code: 400, body: {…}}
value @ ha-automation-editor.ts:737
await in value (async)
handleEvent @ lit-html.ts:2018

fetch-with-auth.ts:20 
        
        
        POST https://xxx/api/config/automation/config/1697489596989 400 (Bad Request)
U @ fetch-with-auth.ts:20
(anonymous) @ hass-call-api.ts:71
callApi @ connection-mixin.ts:155
h @ automation.ts:353
value @ ha-automation-editor.ts:723
handleEvent @ lit-html.ts:2018
ha-automation-editor.ts:737  Uncaught (in promise) {error: 'Response error: 400', status_code: 400, body: {…}}
```

### Additional information

https://community.home-assistant.io/t/message-malformed-integration-not-found-what-have-i-done/178480

### PR DESCRIPTION
**Pull request by codyc1515**
## Proposed change
Adds exception handling for when an unexpected condition occured in the checkValidation function.

This can happen when the automation YAML data becomes malformed, such as when a Device Action entity_id no longer exists or is empty.

Previously, just showed `Automation is unavailable` at the top and nothing about the reason why.

I have modelled the exception handler code off the existing exception handling in the file. Now, shows the error message below `Automation is unavailable` and pops a toast - both are consistent with the behaviour for other kinds of automation issues.

image

## Type of change
- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
- id: '1697489596989'
  alias: Garage - EV Schedule
  description: ''
  trigger:
  - platform: time
    at: '21:01:00'
    id: time_start
    alias: Begin Off Peak (TOD)
  - platform: time
    at: 06:59:00
    id: time_stop
    alias: End Off-Peak (TOD)
  - alias: Begin solar (producing more than 1.8 kW)
    platform: numeric_state
    entity_id:
    - sensor.garage_solar_inverter_pv_power
    for:
      hours: 0
      minutes: 5
      seconds: 0
    id: solar_start
    above: 1.8
  - alias: End solar (producing less 1.8 kW)
    platform: numeric_state
    entity_id:
    - sensor.garage_solar_inverter_pv_power
    for:
      hours: 0
      minutes: 5
      seconds: 0
    below: 1.8
    id: solar_stop_fixed
  - alias: End solar (producing less than EV is consuming)
    platform: numeric_state
    entity_id:
    - sensor.garage_solar_inverter_pv_power
    for:
      hours: 0
      minutes: 5
      seconds: 0
    below: sensor.garage_ev_current_power
    id: solar_stop_dynamic
  condition:
  - condition: numeric_state
    entity_id: zone.home
    above: 0
  action:
  - choose:
    - conditions:
      - condition: trigger
        id:
        - solar_start
        - time_start
      sequence:
      - type: turn_on
        device_id: e9eee6046b80aabf468dc51c90a26297
        entity_id: bef7fc95ab5129fefccd82d61e372e3f
        domain: switch
      - type: turn_on
        device_id: c2064b339e22d70212eeef6834224d89
        entity_id: 6b3f33b06e20e763e58c80373223edd3
        domain: switch
    - conditions:
      - condition: trigger
        id:
        - solar_stop_fixed
        - solar_stop_dynamic
      sequence:
      - type: turn_off
        device_id: e9eee6046b80aabf468dc51c90a26297
        entity_id: bef7fc95ab5129fefccd82d61e372e3f
        domain: switch
      - type: turn_off
        device_id: c2064b339e22d70212eeef6834224d89
        entity_id: 6b3f33b06e20e763e58c80373223edd3
        domain: switch
    - conditions:
      - condition: trigger
        id:
        - time_stop
      sequence:
      - type: turn_off
        device_id: e9eee6046b80aabf468dc51c90a26297
        entity_id: bef7fc95ab5129fefccd82d61e372e3f
        domain: switch
      - type: turn_off
        device_id: c2064b339e22d70212eeef6834224d89
        entity_id: 6b3f33b06e20e763e58c80373223edd3
        domain: switch
  mode: parallel
  max: 10
```

## Additional information

- This PR fixes or closes issue: fixes #19694
- This PR is related to issue or discussion: https://community.home-assistant.io/t/message-malformed-integration-not-found-what-have-i-done/178480
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/automation/ha-automation-editor.ts [hunk 1]
```diff
@@ -476,21 +476,31 @@ export class HaAutomationEditor extends KeyboardShortcutMixin(LitElement) {
...
+        value.valid
+          ? ""
+          : html`${this.hass.localize(
+                `ui.panel.config.automation.editor.${key}s.name`
+              )}:
+              ${value.error}<br />`
+      );
+    } catch (errors: any) {
+      this._validationErrors = errors.message || errors.code;
+      showToast(this, {
```
Comment (bramkragten) [inline src/panels/config/automation/ha-automation-editor.ts:499]: I don't think we should show a toast here? We already display this error message?

Comment (codyc1515) [reply src/panels/config/automation/ha-automation-editor.ts:499]: I wondered this too but that’s how the existing functionality works and I modelled it off that. If we change one then we should change both so they are consistent.

Comment (bramkragten) [reply src/panels/config/automation/ha-automation-editor.ts:499]: I dont see a toast being called in the code outside the catch?

Comment (codyc1515) [reply src/panels/config/automation/ha-automation-editor.ts:499]: It’s more or less identical to this:

https://github.com/home-assistant/frontend/blob/d4386b5e8cc5d354c612fe31a65338629c9b5bb0/src/panels/config/automation/ha-automation-editor.ts#L736

Comment (codyc1515) [reply src/panels/config/automation/ha-automation-editor.ts:499]: That makes sense. I thought not having toast may be an issue on mobile but as it’s only on load, shouldn’t be a problem.

Comment (bramkragten) [reply src/panels/config/automation/ha-automation-editor.ts:499]: Main reason for that (I think) is that on save you can be scrolled down, and can miss the error on the top, on load you will always be at the top of the screen and see the error message.

Comment (codyc1515) [reply src/panels/config/automation/ha-automation-editor.ts:499]: Ah, perfect. I will make that change then.

**Pull request closed without merging**